### PR TITLE
Security Departmental Helmets

### PR DESCRIPTION
**Pull request by Steelpoint**
This PR adds unique helmets that departmental officers can chose to find and wear, the helmets spawn inside the departmental closets 

I've been thinking of ways to give departmental officers something else unique as a fun toy or item for being part of that department. Since giving the Cargo officer a free combat shotgun would be excessive, I've started with this minor idea.

screenshot 2018-11-24 19 52 41
Cargo-Engineering-Medical-Science

The helmets themselves are cosmetic only, while I'd like to explore them having minor differences (Med helm with a bit more bio prot, sci with a bit more explosive prot, etc) but I want to avoid wars over sec buffs.

All the helmets can mount a Seclight.

Feedback appreciated, and if there is anything cool I could look to further giving departmental officers as a neat toy or minor item. 

:cl: Steelpoint
add: Departmental Security Officers have been issued new helmets for their assigned department, each helmet is only cosmetically different. They can be found inside the departmental lockers.
/:cl:

### CODE REVIEW

**Comment by Steelpoint** [comment]
I might do a slight rework on the Sci and Eng helmets to be more akin to the Cargo helm.

**Comment by Grazyn** [comment]
You may want to make the helmet strip a bit wider so it doesn't look like the tip of a penis

**Comment by Steelpoint** [comment]
screenshot 2018-11-23 21 52 19

I've updated the side sprites a bit.

**Comment by subject217** [comment]
Kor had a good PR for unique departmental sec guard stuff from like April or thereabouts. Might be a good place to look for some inspiration.

**Comment by Steelpoint** [comment]
Yeah but some of his proposals were way too good, cargo officer gets a Combat Shotgun or a traitor item  Sci officer gets reactive armour or a advance gun.

I might look at something more mild like the med officer getting a first aid kit and akin

**Comment by Steelpoint** [comment]
The mask implication is its blast plating to help protect against fragmentation and debris from explosions.

**Comment by Steelpoint** [comment]
I think I'll do one more sprite edit to standardise the sprite dimensions of the departmental colours so the positioning is consistent.

**Comment by Steelpoint** [comment]
I've done a final update on the sprites, and have updated the OP with the latest varient.

screenshot 2018-11-24 19 52 41
Cargo-Engineering-Medical-Science

I'll leave this PR as is. I'll consider adding a minor item of note to each of the dept sec closets in a future PR.

**Comment by improvedname** [comment]
Remember that mini hypospray steel give that to the medbay officer with safeties

File: code/modules/clothing/head/helmet.dm [hunk 1]
```diff
@@ -69,6 +69,36 @@
 			return
 	return ..()
 
+/obj/item/clothing/head/helmet/med
+	name = "helmet (med)"
+	desc = "Standard Security gear. Has Medbay's symbol printed on the side."
+	icon_state = "helmet-med"
+	item_state = "helmet-med"
+	can_flashlight = TRUE
```
Comment (Cyberboss) [inline code/modules/clothing/head/helmet.dm]:
```suggestion
```

File: code/modules/clothing/head/helmet.dm [hunk 2]
```diff
@@ -69,6 +69,36 @@
...
+	item_state = "helmet-med"
+	can_flashlight = TRUE
+
+/obj/item/clothing/head/helmet/sci
+	name = "helmet (sci)"
+	desc = "Standard Security gear. Has extra blast plating covering the mouth area."
+	flags_inv = HIDEMASK|HIDEHAIR|HIDEEARS
+	icon_state = "helmet-sci"
+	item_state = "helmet-sci"
+	can_flashlight = TRUE
```
Comment (Cyberboss) [inline code/modules/clothing/head/helmet.dm]:
```suggestion
```

File: code/modules/clothing/head/helmet.dm [hunk 3]
```diff
@@ -69,6 +69,36 @@
...
+
+/obj/item/clothing/head/helmet/eng
+	name = "helmet (eng)"
+	desc = "Standard Security gear. Has Engineering symbol printed on the side, and the visor is slightly tinted."
+	icon_state = "helmet-eng"
+	item_state = "helmet-eng"
+	can_flashlight = TRUE
```
Comment (Cyberboss) [inline code/modules/clothing/head/helmet.dm]:
```suggestion
```

File: code/modules/clothing/head/helmet.dm [hunk 4]
```diff
@@ -69,6 +69,36 @@
...
+
+/obj/item/clothing/head/helmet/crg
+	name = "helmet (crg)"
+	desc = "Standard Security gear. Has cargo colouring on the edges."
+	icon_state = "helmet-crg"
+	item_state = "helmet-crg"
+	can_flashlight = TRUE
```
Comment (Cyberboss) [inline code/modules/clothing/head/helmet.dm]:
```suggestion
```

File: code/modules/clothing/head/helmet.dm [hunk 5]
```diff
@@ -69,6 +69,36 @@
 			return
 	return ..()
 
+/obj/item/clothing/head/helmet/med
```
Comment (Cyberboss) [inline code/modules/clothing/head/helmet.dm]:
```suggestion
/obj/item/clothing/head/helmet/sec/med
```

File: code/modules/clothing/head/helmet.dm [hunk 6]
```diff
@@ -69,6 +69,36 @@
...
+
+/obj/item/clothing/head/helmet/sci
+	name = "helmet (sci)"
+	desc = "Standard Security gear. Has extra blast plating covering the mouth area."
+	flags_inv = HIDEMASK|HIDEHAIR|HIDEEARS
+	icon_state = "helmet-sci"
+	item_state = "helmet-sci"
+	can_flashlight = TRUE
+
+/obj/item/clothing/head/helmet/eng
```
Comment (Cyberboss) [inline code/modules/clothing/head/helmet.dm]:
```suggestion
/obj/item/clothing/head/helmet/sec/eng
```

File: code/modules/clothing/head/helmet.dm [hunk 7]
```diff
@@ -69,6 +69,36 @@
...
+	name = "helmet (eng)"
+	desc = "Standard Security gear. Has Engineering symbol printed on the side, and the visor is slightly tinted."
+	icon_state = "helmet-eng"
+	item_state = "helmet-eng"
+	can_flashlight = TRUE
+
+/obj/item/clothing/head/helmet/crg
```
Comment (Cyberboss) [inline code/modules/clothing/head/helmet.dm]:
```suggestion
/obj/item/clothing/head/helmet/sec/crg
```

File: code/modules/clothing/head/helmet.dm [hunk 8]
```diff
@@ -69,6 +69,36 @@
...
 	return ..()
 
+/obj/item/clothing/head/helmet/med
+	name = "helmet (med)"
+	desc = "Standard Security gear. Has Medbay's symbol printed on the side."
+	icon_state = "helmet-med"
+	item_state = "helmet-med"
+	can_flashlight = TRUE
+
+/obj/item/clothing/head/helmet/sci
```
Comment (Cyberboss) [inline code/modules/clothing/head/helmet.dm]:
```suggestion
/obj/item/clothing/head/helmet/sec/sci
```

File: code/game/objects/structures/crates_lockers/closets/secure/security.dm [hunk 9]
```diff
@@ -142,27 +142,31 @@
...
 	..()
+	new /obj/item/clothing/head/helmet/sci(src)
 	new /obj/item/clothing/accessory/armband/science(src)
 	new /obj/item/encryptionkey/headset_sci(src)
 
 /obj/structure/closet/secure_closet/security/med
 
 /obj/structure/closet/secure_closet/security/med/PopulateContents()
 	..()
+	new /obj/item/clothing/head/helmet/med(src)
```
Comment (Cyberboss) [inline code/game/objects/structures/crates_lockers/closets/secure/security.dm]:
```suggestion
	new /obj/item/clothing/head/helmet/sec/med(src)
```

File: code/game/objects/structures/crates_lockers/closets/secure/security.dm [hunk 10]
```diff
@@ -142,27 +142,31 @@
...
 	..()
+	new /obj/item/clothing/head/helmet/eng(src)
 	new /obj/item/clothing/accessory/armband/engine(src)
 	new /obj/item/encryptionkey/headset_eng(src)
 
 /obj/structure/closet/secure_closet/security/science
 
 /obj/structure/closet/secure_closet/security/science/PopulateContents()
 	..()
+	new /obj/item/clothing/head/helmet/sci(src)
```
Comment (Cyberboss) [inline code/game/objects/structures/crates_lockers/closets/secure/security.dm]:
```suggestion
	new /obj/item/clothing/head/helmet/sec/sci(src)
```

File: code/game/objects/structures/crates_lockers/closets/secure/security.dm [hunk 11]
```diff
@@ -142,27 +142,31 @@
...
 	..()
+	new /obj/item/clothing/head/helmet/crg(src)
 	new /obj/item/clothing/accessory/armband/cargo(src)
 	new /obj/item/encryptionkey/headset_cargo(src)
 
 /obj/structure/closet/secure_closet/security/engine
 
 /obj/structure/closet/secure_closet/security/engine/PopulateContents()
 	..()
+	new /obj/item/clothing/head/helmet/eng(src)
```
Comment (Cyberboss) [inline code/game/objects/structures/crates_lockers/closets/secure/security.dm]:
```suggestion
	new /obj/item/clothing/head/helmet/sec/eng(src)
```

File: code/game/objects/structures/crates_lockers/closets/secure/security.dm [hunk 12]
```diff
@@ -142,27 +142,31 @@
 
 /obj/structure/closet/secure_closet/security/cargo/PopulateContents()
 	..()
+	new /obj/item/clothing/head/helmet/crg(src)
```
Comment (Cyberboss) [inline code/game/objects/structures/crates_lockers/closets/secure/security.dm]:
```suggestion
	new /obj/item/clothing/head/helmet/sec/crg(src)
```

**Review by Cyberboss — CHANGES_REQUESTED** [review]
Any reason you aren't overriding the basic sec helmet?

**Comment by Steelpoint** [comment]
I missed that, thanks @Cyberboss 

Changed effected.

**Comment by Memager** [comment]
The cargo helmet is almost the same color as the engineering one, maybe use the orange from the engi armband?

**Comment by Steelpoint** [comment]
Done

File: code/modules/clothing/head/helmet.dm [hunk 13]
```diff
@@ -69,6 +69,28 @@
...
+	flags_inv = HIDEMASK|HIDEHAIR|HIDEEARS
+	icon_state = "helmet-sci"
+	item_state = "helmet-sci"
+
+/obj/item/clothing/head/helmet/sec/eng
+	desc = "Standard Security gear. Has Engineering symbol printed on the side, and the visor is slightly tinted."
+	icon_state = "helmet-eng"
+	item_state = "helmet-eng"
+
+/obj/item/clothing/head/helmet/sec/crg
```
Comment (ninjanomnom) [inline code/modules/clothing/head/helmet.dm]: crg? We don't live in a world of limited name sizes, make it more clear

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

**Comment by Cyberboss** [comment]
Your PR is getting old. Either Travis CI is failing, it is still labelled work in progress (change this by adding `[READY]` to the start of your title), or there are unaddressed change requests. This may be closed soon to free up space on our tracker if it remains idle. In the event that it is, please ask in #coderbus on Rizon for it to be reopened once you have resumed work on it.

**Comment by Steelpoint** [comment]
Not enough interest.

**Pull request closed without merging**